### DISCUSSION TITLE
Error parsing config file when i using filter

### DISCUSSION THREAD

**Discussion by dimkene** [prometheus/snmp_exporter#980] [Q&A] [answered]
I want to filter data by interface name, but when I add the line with the filter configuration taken from https://github.com/prometheus/snmp_exporter/blob/main/generator/README.md to the config, I get an error in the docker logs
```
Error parsing config file" err="yaml: unmarshal errors:\n  line 1370: cannot unmarshal !!map into []config.DynamicFilter
Possible old config file, see https://github.com/prometheus/snmp_exporter/blob/main/auth-split-migration.md
```
But in auth-split-migration.md I did not find any information that the filter section was changed

**Comment by candlerb** [comment, marked as answer]
There's something broken in your snmp.yml around line 1370 - but you haven't shown that file.

"Possible old config file" is reported for any problems in snmp.yml, since this is a common problem.  But other problems are possible.

Note that for any significant changes you're not supposed to edit snmp.yml anyway; you're supposed to edit generator.yml and then run the generator on it.

**Reply by dimkene** [reply]
Thanks, when using the generator, snmp_exporter correctly reads the config
